docs/user_guide/method/method.rst

```diff
@@ -0,0 +1,2384 @@
+understood by anyone familiar with the underlying physical concepts, the
+organization of the :ref:`method_api` object in **mrsimulator** and its related
+composite objects require a more detailed explanation of their design. This
+section assumes that you are already familiar with the topics covered in the
+introduction sections :ref:`getting_started`,
+:ref:`introduction_isotopomers_example`, and :ref:`fitting_example`.
+
+
+
+Overview
+--------
+
+An experimental NMR method involves a sequence of rf pulses, free evolution
+periods, and sample motion. The **Method** object in **mrsimulator** models the
+spectrum from an NMR pulse sequence. The **Method** object is designed to be
+versatile in its ability to model spectra from various multi-pulse NMR methods
+using concepts from the `symmetry pathway approach
+<https://doi.org/10.1016/j.pnmrs.2010.11.003>`_ where a pulse sequence is
+understood in terms of a set of desired (and undesired)  *transition pathways*.
+Each transition pathway is associated with a single resonance in a
+multi-dimensional NMR spectrum. The transition pathway signal encodes
+information about the spin system interactions in its amplitude and correlated
+frequencies. Consider the illustration of a 2D pulse sequence shown below, where
+a desired signal for the method is associated with a particular transition
+pathway, :math:`{\hat{A} \rightarrow \hat{B} \rightarrow \hat{C} \rightarrow
+\hat{D} \rightarrow \hat {E} \rightarrow \hat{F}}`.
+
+.. figure:: ../../_static/TransitionPathway.*
+    :width: 600
+    :alt: figure
+    :align: center
+
+Here, the first spectral dimension, i.e., the Fourier transform of the
+transition pathway signal as a function of :math:`t_1`, derives its *average
+frequency*, :math:`\overline{\Omega}_1`, from a weighted average of the
+:math:`\hat{A}`, :math:`\hat{B}`, and :math:`\hat{C}` transition frequencies.
+The second spectral dimension, i.e., the FT with respect to :math:`t_2`, derives
+its average frequency, :math:`\overline{\Omega}_2`, from a weighted average of
+the :math:`\hat{E}`, and :math:`\hat{F}` transition frequencies. Much of the
+experimental design and implementation of an NMR method is in identifying the
+desired transition pathways and finding ways to acquire their signals while
+eliminating all undesired transition pathway signals.
+
+While NMR measurements occur in the time domain, **mrsimulator** simulates the
+corresponding multi-dimensional spectra directly in the frequency domain. The
+**Method** object in **mrsimulator** needs only a few details of the NMR pulse
+sequence to generate the spectrum. It mimics the result of the pulse sequence
+given the desired transition pathways and their complex amplitudes and average
+frequencies in each spectroscopic dimension of the dataset. To this end, a
+**Method** object is organized according to the UML diagram below.
+
+.. figure:: ../../_static/MethodUML.*
+    :width: 700
+    :alt: figure
+    :align: center
```

Comment:
@pjgrandinetti ``freq_contrib`` list can take a list of 0 to 9 **FrequencyEnum** objects. The UML diagram needs to be updated.

Also the **TotalMixingQuery** object does not exist. It is an enumeration passed to the query attribute of a mixing event.

**MixingEvent** has a ``query`` attribute now. Was changed a few weeks ago

Reply:
okay, I'll fix the UML diagram.

If TotalMixQuery can be placed in ch1, ch2, or ch3, then it belongs in the UML diagram.    This is why I was asking about a parent class for Mix query, that RotationalQuery and TotalMixQuery would inherit from.

Reply:
A better attribute would be instances/attributes (still not 100% on enum language) from the **MixingEnum** class since can also have ``NoMixing``.

And then there are strings for the enumerations too! This might be something better do discuss during our meeting tomorrow...

Reply:
yes, let's discuss this at the meeting.  The UML is not so critical in the docs right now, but we'll need to get it right for the manuscript.

Reply:
UML will be updated in next PR